advanced-cache.php

```diff
@@ -455,12 +455,13 @@ function add_debug_html_to_output( $debug_html ) {
 
 // Get the batcache
 $batcache->cache = wp_cache_get($batcache->key, $batcache->group);
+$has_expired = ! empty( $batcache->cache['time'] ) && time() > $batcache->cache['time'] + $batcache->cache['max_age'];
 
 if ( isset( $batcache->cache['version'] ) && $batcache->cache['version'] != $batcache->url_version ) {
 	// Always refresh the cache if a newer version is available.
 	$batcache->do = true;
-} else if ( $batcache->seconds < 1 || $batcache->times < 2 ) {
-	// Are we only caching frequently-requested pages?
+} else if ( $has_expired && ( $batcache->seconds < 1 || $batcache->times < 2 ) ) {
```

Comment:
In case the `$has_expired` is `false`, then the code is going to skip to the `else` no matter what the values of `$batcache->seconds` and `$batcache->times` is. So there is not going to be any special handling for those extreme values ( `< 1` and `< 2` ).

Which means, there is going to be some overhead with `$batcache->req_key` memcache operations ( `increase` and `delete` ), which is not really needed, IMHO.

So I believe that we should move the `$has_expired` check to inside the conditional, and have `if...else` in there, with `do` being set to `true` or `false` appropriately.

Reply:
IIUC, `$batcache->do = $has_expired` would do it